Can't run `mdbook build` locally without installing or disabling linkcheck

As of #2103:

```text
$ mdbook build --open
2024-12-30 11:30:43 [INFO] (mdbook::book): Book building has started
2024-12-30 11:30:43 [INFO] (mdbook::book): Running the html backend
2024-12-30 11:30:44 [INFO] (mdbook::book): Running the linkcheck backend
2024-12-30 11:30:44 [INFO] (mdbook::renderer): Invoking the "linkcheck" renderer
Checking files changed in master...: 
exec mdbook-linkcheck2 -f 
/Users/stuart/Dev/rust/dev-guide/ci/linkcheck.sh: line 47: exec: mdbook-linkcheck2: not found
2024-12-30 11:30:44 [WARN] (mdbook::renderer): Error writing the RenderContext to the backend, Broken pipe (os error 32)
2024-12-30 11:30:44 [ERROR] (mdbook::renderer): Renderer exited with non-zero return code.
2024-12-30 11:30:44 [ERROR] (mdbook::utils): Error: Rendering failed
2024-12-30 11:30:44 [ERROR] (mdbook::utils): 	Caused By: The "linkcheck" renderer failed
```

---

Building the book now fails if `mdbook-linkcheck2` has not been manually installed, even if the contributor doesn't care about linkcheck and just wants to build the book to preview their changes.

And while there is a workaround available, it isn't mentioned by the error message at all.

---

As a temporary workaround, this works:

```
$ SKIP_LINKCHECK=1 mdbook build
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -57,7 +57,7 @@ jobs:
           cargo install mdbook-mermaid --version ${{ env.MDBOOK_MERMAID_VERSION }}
 
       - name: Check build
-        run: mdbook build
+        run: ENABLE_LINKCHECK=1 mdbook build
 
       - name: Deploy to gh-pages
         if: github.event_name == 'push'

--- a/README.md
+++ b/README.md
@@ -56,12 +56,14 @@ The build files are found in the `book/html` directory.
 
 ### Link Validations
 
-We use `mdbook-linkcheck2` to validate URLs included in our documentation.
-`linkcheck` will be run automatically when you build with the instructions in the section above.
+We use `mdbook-linkcheck2` to validate URLs included in our documentation. Link
+checking is **not** run by default locally, though it is in CI. To enable it
+locally, set the environment variable `ENABLE_LINKCHECK=1` like in the
+following example.
 
-> [!NOTE]
-> The link validation can be skipped by setting the following environment variable:
-> `SKIP_LINKCHECK=1 mdbook ...`
+```console
+$ ENABLE_LINKCHECK=1 mdbook serve
+```
 
 ### Table of Contents
 

--- a/ci/linkcheck.sh
+++ b/ci/linkcheck.sh
@@ -9,7 +9,7 @@ set_github_token() {
   jq '.config.output.linkcheck."http-headers"."github\\.com" = ["Authorization: Bearer $GITHUB_TOKEN"]'
 }
 
-if [ ! -z "$SKIP_LINKCHECK" ] ; then
+if [ -z "$ENABLE_LINKCHECK" ] ; then
   echo "Skipping link check."
   exit 0
 fi

